src/utils/proton.js

```diff
@@ -48,14 +48,15 @@ class ProtonSDK {
     const savedUserAuth = JSON.parse(localStorage.getItem('savedUserAuth-foobar'));
     if (savedUserAuth) {
       try {
-        this.link = await ConnectWallet({
-          linkOptions: { chainId: this.chainId, endpoints: this.endpoints },
-          transportOptions: { requestAccount: this.requestAccount, backButton: true },
-          selectorOptions: { appName: this.appName, appLogo: Foobar, showSelector: false }
+        const { link, session } = await ConnectWallet({
```

Comment:
Instead of having ConnecWallet code in multiple places, refactor it into a new function that takes in restoreSession as a parameter

Reply:
will do! :)